src/libraries/System.ComponentModel.TypeConverter/tests/Design/DesigntimeLicenseContextSerializerTests.cs

```diff
@@ -0,0 +1,146 @@
+            }
+        }
+
+        [ConditionalTheory(nameof(AreBinaryFormatterAndRemoteExecutorSupportedOnThisPlatform))]
+        [InlineData("key")]
+        [InlineData("")]
+        public static void SerializeWithBinaryFormatter_DeserializeWithBinaryWriter(string key)
+        {
+            AppContext.SetSwitch(enableBinaryFormatter, true);
+            AppContext.SetSwitch(enableBinaryFormatterInTypeConverter, true);
+            var context = new DesigntimeLicenseContext();
+            context.SetSavedLicenseKey(typeof(int), key);
+            var assembly = typeof(DesigntimeLicenseContextSerializer).Assembly;
+            using (MemoryStream stream = new MemoryStream())
+            {
+                long position = stream.Position;
+                DesigntimeLicenseContextSerializer.Serialize(stream, key, context);
+                stream.Seek(position, SeekOrigin.Begin);
+                VerifyStreamFormatting(stream);
+
+                string tempPath = Path.GetTempPath();
+                using (FileStream outStream = File.Create(Path.Combine(tempPath, "_temp_SerializeWithBinaryFormatter_DeserializeWithBinaryWriter")))
+                {
+                    stream.Seek(position, SeekOrigin.Begin);
+                    stream.CopyTo(outStream);
+                }
+            }
+
+            RemoteExecutor.Invoke((key) =>
+            {
+                var assembly = typeof(DesigntimeLicenseContextSerializer).Assembly;
+                Type runtimeLicenseContextType = assembly.GetType("System.ComponentModel.Design.RuntimeLicenseContext");
+                Assert.NotNull(runtimeLicenseContextType);
+                object runtimeLicenseContext = Activator.CreateInstance(runtimeLicenseContextType);
+                Assert.NotNull(runtimeLicenseContext);
+                FieldInfo _savedLicenseKeys = runtimeLicenseContextType.GetField("_savedLicenseKeys", BindingFlags.NonPublic | BindingFlags.Instance);
+                Assert.NotNull(_savedLicenseKeys);
+                _savedLicenseKeys.SetValue(runtimeLicenseContext, new Hashtable());
+
+                Type designtimeLicenseContextSerializer = assembly.GetType("System.ComponentModel.Design.DesigntimeLicenseContextSerializer");
+                Assert.NotNull(designtimeLicenseContextSerializer);
+                MethodInfo deserializeMethod = designtimeLicenseContextSerializer.GetMethod("Deserialize", BindingFlags.NonPublic | BindingFlags.Static);
+                Assert.NotNull(deserializeMethod);
+
+                string tempPath = Path.GetTempPath();
+                using (FileStream stream = File.Open(Path.Combine(tempPath, "_temp_SerializeWithBinaryFormatter_DeserializeWithBinaryWriter"), FileMode.Open))
+                {
+                    try
+                    {
+                        deserializeMethod.Invoke(null, new object[] { stream, key, runtimeLicenseContext });
+                    }
+                    catch (TargetInvocationException exception)
```

Comment:
What happens if the above doesn't throw? Use `Assert.Throws` instead.

Reply:
I can't get the base exception using Assert.Throws right? 

I did change things here to catch all exceptions appropriately though.

Reply:
Here's an example of how to do this:

https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Text.RegularExpressions/tests/GroupCollectionTests2.cs#L176-L177